src/components/CategoryList/CategoryList.js

```diff
@@ -0,0 +1,53 @@
+import React, { useState, useEffect } from 'react';
+import './CategoryList.scss';
+
+const CategoryList = props => {
+  const [mainNameList, setMainNameList] = useState([]);
+  const [currentId, setCurrentId] = useState(0);
+
+  useEffect(() => {
+    fetch('/data/categoryList.json', {
+      method: 'GET',
+    })
+      .then(res => res.json())
+      .then(data => {
+        setMainNameList(data);
+      });
+  }, []);
+
+  return (
+    <div className="navCategoryWrap">
+      <div className="categoryList">
+        <ul className="categoryListMain">
+          {mainNameList.map(({ id, icon, mainname, subcategory }) => {
+            return (
+              <li
+                className="mainCategory"
+                key={id}
+                data={mainname}
+                onMouseEnter={() => setCurrentId(id)}
```

Comment:
이런 callback은 커링을 이용하면 정말 이쁘게 표현 할 수 있습니다.

```jsx
const onMouseEnterTab = (id) => (e) => setCurrentId(id);
```
```suggestion
                onMouseEnter={setCurrentId(id)}
```

callback 함수는 들어오는 argument와 parameter가 같다면 생략이 가능하다는 점을 이용한 겁니다.

이걸 참고하면 이해가 될 것 같습니다. 
[참고자료](https://www.youtube.com/watch?v=dhrXFrrkYf0)